src/Microsoft.Build.UnGAC/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.UnGAC
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                List<string> allInstancesOfGACUtil = new List<string>();
+                StreamReader output;
+                StreamWriter input;
+
+                Process proc = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = "C:\\Windows\\System32\\cmd.exe",
+                        Arguments = "/k", // Keep cmd open until we're done
+                        RedirectStandardOutput = true,
+                        RedirectStandardInput = true,
+                        UseShellExecute = false
+                    }
+                };
+                proc.Start();
+
+                // Hook into standard input/output of the process
+                output = proc.StandardOutput;
+                input = proc.StandardInput;
+
+                if (!input.BaseStream.CanWrite)
+                {
+                    proc.Close();
+                    proc.Dispose();
+                    Console.WriteLine("Could not use where.exe to find gacutil.exe. Exiting...");
+                    return;
+                }
+
+                // Use where.exe to find gacutil.exe
+                input.WriteLine("where /F /R \"C:\\Program Files (x86)\\Microsoft SDKs\\Windows\" gacutil.exe");
+
+                // Throw away the first line - it's the command we just used
+                output.ReadLine();
```

Comment:
This would probably make it messier rather than cleaner, but you should be able to set echo to off to suppress this.